src/legacy/core_plugins/kibana/public/management/sections/objects/lib/resolve_saved_objects.js

```diff
@@ -140,19 +140,38 @@ export async function resolveIndexPatternConflicts(
   overwriteAll
 ) {
   let importCount = 0;
+
   await awaitEachItemInParallel(conflictedIndexPatterns, async ({ obj }) => {
+    // Resolve search index reference:
     let oldIndexId = obj.searchSource.getOwnField('index');
     // Depending on the object, this can either be the raw id or the actual index pattern object
     if (typeof oldIndexId !== 'string') {
       oldIndexId = oldIndexId.id;
     }
-    const resolution = resolutions.find(({ oldId }) => oldId === oldIndexId);
+    let resolution = resolutions.find(({ oldId }) => oldId === oldIndexId);
+    if (resolution) {
+      const newIndexId = resolution.newId;
+      await obj.hydrateIndexPattern(newIndexId);
+    }
+
+    // Resolve filter index reference:
+    const filter = (obj.searchSource.getOwnField('filter') || []).map((filter) => {
+      if (!(filter.meta && filter.meta.index)) {
+        return filter;
+      }
+
+      resolution = resolutions.find(({ oldId }) => oldId === filter.meta.index);
+      return resolution ? ({ ...filter, ...{ meta: { ...filter.meta, index: resolution.newId } } }) : filter;
```

Comment:
also not super clear why `management` changes saved object internal content and mutate it

Reply:
Yeah I'm not sure if this was implemented before we had migrations, but we're essentially performing migrations in the frontend which is rather brittle :( We use this code path when importing JSON files, but newer exports are made in NDJSON and if you importing NDJSON we'll do all the migrations server-side with a much cleaner architecture. Unfortunately JSON support will probably have to stay till 8.0.0.